addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -53,32 +55,165 @@ var SlideDialog = Widget.extend({
+            _.extend(values, {
+                'image': canvas.toDataURL().split(',')[1],
+                'slide_type': canvas.height > canvas.width ? 'document' : 'presentation',
+                'mime_type': this.file.type,
+                'datas': this.file.data
+            });
+        }
+        if (/^image\/.*/.test(this.file.type)) {
+            _.extend(values, {
+                'slide_type': 'infographic',
+                'mime_type': this.file.type === 'image/svg+xml' ? 'image/png' : this.file.type,
+                'datas': this.file.type === 'image/svg+xml' ? this._svgToPng() : this.file.data
+            });
+        }
+        return values;
+    },
+    /**
+     * @private
+     */
+    _fileReset: function () {
         var control = this.$('#upload');
         control.replaceWith(control = control.clone(true));
         this.file.name = false;
     },
+
+    _getModalButtons: function () {
+        var btn_list = [];
+        var state = this.get('state');
+        if (state === '_select') {
+            btn_list.push({text: _t("Cancel"), classes: 'o_w_slide_cancel', close: true});
+        } else if (state !== '_upload') { // no button when uploading
+            if (this.canUpload) {
+                if (this.canPublish) {
+                    btn_list.push({text: _t("Save as Draft"), classes: 'o_w_slide_upload', click: this._onClickFormSubmit.bind(this)});
+                    btn_list.push({text: _t("Publish"), classes: 'btn-primary o_w_slide_upload o_w_slide_upload_published', click: this._onClickFormSubmit.bind(this)});
+                } else {
+                    btn_list.push({text: _t("Save as Draft"), classes: 'btn-primary o_w_slide_upload', click: this._onClickFormSubmit.bind(this)});
+                }
+            }
+            btn_list.push({text: _t("Go Back"), classes: 'o_w_slide_go_back', click: this._onClickGoBack.bind(this)});
+        }
+        return btn_list;
+    },
     /**
+     * Get value for category_id and tag_ids (ORM cmd) to send to server
+     *
      * @private
-     * @param {string} message
      */
-    _displayAlert: function (message) {
-        this.$('.alert-warning').remove();
-        $('<div class="alert alert-warning" role="alert">' + message + '</div>').insertBefore(this.$('form'));
+    _getSelect2DropdownValues: function (){
+        var result = {};
+        // tags
+        var tags_values = [];
+        _.each($('#tag_ids').select2('data'), function (val) {
+            if (val.create) {
+                tags_values.push([0, 0, {'name': val.text}]);
+            } else {
+                tags_values.push([4, val.id]);
+            }
+        });
+        if (tags_values) {
+            result['tag_ids'] = tags_values;
+        }
+        // category
+        var category_value = $('#category_id').select2('data');
+        if (category_value && category_value.create) {
+            result['category_id'] = [0, {'name': category_value.text}];
+        } else {
```

Comment:
Would be easier to read if it was `else if(categoryValue)` and then avoid ternary check but I don't know it we have to specifically set "category_id" to "null" ? Let me know if it's not clear.